runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractQueuedStreamView.java

```diff
@@ -397,6 +401,34 @@ private void processTrimmedException(TrimmedException te) {
                 .collect(Collectors.toList());
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<ILogData> remainingAtMost(int maxEntries) {
+        // TODO: instrument remainingUpTo directly to limit the number of
+        // of entries returned instead of calling it.
+        if (maxEntries < 1) {
+            throw new IllegalArgumentException(
+                    "remainingAtMost(): maxEntries cannot be less than 1.");
+        }
+
+        long maxGlobal = getMaxGlobalFromMaxEntries(maxEntries);
+        if (Address.nonAddress(maxGlobal)) {
+            return Collections.emptyList();
+        }
```

Comment:
This will emit all of of unnecessary sequencer queries

Reply:
I think our maxResolution is broken as of now.